source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.cc

```diff
@@ -0,0 +1,130 @@
+#include "extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.h"
+
+#include <memory>
+
+namespace Envoy {
+namespace Quic {
+
+QuicFilterManagerConnectionImpl::QuicFilterManagerConnectionImpl(
+    std::unique_ptr<EnvoyQuicConnection> connection, Event::Dispatcher& dispatcher)
+    : quic_connection_(std::move(connection)), filter_manager_(*this), dispatcher_(dispatcher),
+      stream_info_(dispatcher.timeSource()) {
+  // TODO(danzh): Use QUIC specific enum value.
+  stream_info_.protocol(Http::Protocol::Http2);
+}
+
+void QuicFilterManagerConnectionImpl::addWriteFilter(Network::WriteFilterSharedPtr filter) {
+  filter_manager_.addWriteFilter(filter);
+}
+
+void QuicFilterManagerConnectionImpl::addFilter(Network::FilterSharedPtr filter) {
+  filter_manager_.addFilter(filter);
+}
+
+void QuicFilterManagerConnectionImpl::addReadFilter(Network::ReadFilterSharedPtr filter) {
+  filter_manager_.addReadFilter(filter);
+}
+
+bool QuicFilterManagerConnectionImpl::initializeReadFilters() {
+  return filter_manager_.initializeReadFilters();
+}
+
+void QuicFilterManagerConnectionImpl::addConnectionCallbacks(Network::ConnectionCallbacks& cb) {
+  network_connection_callbacks_.push_back(&cb);
+}
+
+void QuicFilterManagerConnectionImpl::addBytesSentCallback(
+    Network::Connection::BytesSentCb /*cb*/) {
+  // TODO(danzh): implement to support proxy. This interface is only called from
```

Comment:
Should this be NOT_IMPLEMENTED here and elsewhere in this file?

Reply:
This is only called in TcpProxy::Filter::onUpstreamEvent().  I think it's not supposed to be reached in quic.

Same reason for all the ASSERT in this file. They are supposed to be NOT_REACHED, but since NOT_REACHED crashes in production, we changed it to ASSERT in the old PR in case of bad config.

Reply:
I would prefer that we either crash hard if things should not happen, or actually implement stuff. Since QUIC is not production ready yet, can we just make everything that shouldn't happen NOT_IMPLEMENTED for now? Same elsewhere?

Reply:
Done. Anything shouldn't happen -> NOT_REACHED, anything should be implemented to let integration test pass -> NOT_IMPLEMENTED, anything that we don't need in a hurry -> TODO.